CONTRIBUTING.md

```diff
@@ -35,6 +35,34 @@ The Android-specific library AAR is built with:
 
 (The `ANDROID_HOME` environment variable must be set appropriately.)
 
+## Adding a New Network Engine Implementation
+
+Currently, `ably-java` supports two different engines for network operations (HTTP calls and WebSocket connections):
+
+- **Default Engine**: Utilizes the built-in `HttpUrlConnection` for HTTP calls and the TooTallNate/Java-WebSocket library for WebSocket connections.
+- **OkHttp Engine**: Utilizes the OkHttp library for both HTTP and WebSocket connections.
+
+These engines are designed to be swappable. By default, the library comes with the default engine, but you can easily replace it with the OkHttp engine:
```

Comment:
Can we use word `websocket library` instead of `engine`. We might like to keep `engine` term only for technical impl. 
It can get confusing for users wdyt